src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -1811,10 +1814,14 @@ module Crystal
         filter_vars cond_type_filters, &.not
       when Or
         # Try to apply boolean logic: `!(a || b)` is `!a && !b`
+        cond_left = cond.left
```

Comment:
Could it be a good case for refactoring? Making a helper method along the lines of:

```cr
def single_expression(cond)
  if cond.is_a?(Expressions) && cond.expressions.size == 1
    return cond[0]
  end
end

# then you could re-use it like:
cond_left = single_expression(cond) || cond.left

# and l8r on, around line 2116
when Expressions
  return unless cond = single_expression(node)
  return get_while_cond_assign_target(cond)
```